File: frontend-mealsonwheels/src/types/order/OrderMealDTO.ts
// src/types/order/OrderMealDTO.ts

export interface OrderMealDTO {
  mealId: number;
  mealName: string;
  mealPhoto: Uint8Array | null; // Can also be base64 string depending on frontend handling
  mealDesc: string;
  mealType: string;
  mealDietary: string;
  quantity: number;
  mealCreatedDate: string; // ISO string
}

File: backend-mealsonwheels/src/main/java/com/merrymeal/mealsonwheels/dto/order/OrderMealDTO.java
package com.merrymeal.mealsonwheels_backend.dto;

import lombok.*;

import java.time.LocalDateTime;

import com.merrymeal.mealsonwheels_backend.dto.mealDTOs.MealDTO;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class OrderMealDTO {

    private Long mealId;
    private String mealName;
    private byte[] mealPhoto;
    private String mealDesc;
    private String mealType;
    private String mealDietary;
    private Integer quantity;
    private LocalDateTime mealCreatedDate;

    public OrderMealDTO(MealDTO meal, Integer quantity) {
        this.mealId = meal.getId();
        this.mealName = meal.getMealName();
        this.mealPhoto = meal.getMealPhoto();
        this.mealDesc = meal.getMealDesc();
        this.mealType = meal.getMealType();
        this.mealDietary = meal.getMealDietary();
        this.mealCreatedDate = meal.getMealCreatedDate();
        this.quantity = quantity;
    }
}
